Microsoft Azure transcription service refactoring

wsmirnow opened the pull request:
The previous implementation of the Azure transcription service uses the Microsoft Speech SDK. On Linux systems it needs gstreamer installed as it will be used to extract and push audio stream to Azure. It comes with some complications. First of all, a memory leak. But also other issues we've found in production. This was the reason to refactor the code using the Azure Speech Services endpoints directly. 

Microsoft Azure Speech Services provide multiple ways to transcribe audio for specific use cases. As we want to transcript an static, not live, audio stream, the best option is the [Batch API](https://learn.microsoft.com/en-us/azure/ai-services/speech-service/batch-transcription). It works like transcription services from other vendors

- Upload audio track
- Poll transcription job status
- Download the result on success
- Start an attach-transcript workflow

To test the module you need an Microsoft account with at least a standard tear subscription. Free subscription does not work as documented [here](https://learn.microsoft.com/en-us/azure/ai-services/speech-service/batch-transcription-create#prerequisites). Initially the code was written for Opencast 14. Since then we run it for two semesters in production. For the contribution it was rebased to develop and added the missing documentation. Some minor changes to handling captions (see [documentation](https://docs.opencast.org/develop/admin/#configuration/subtitles/)) are also applied.


### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

wsmirnow committed:
Azure transcription service should wait for running workflow

Wait to start attach weokflow if another workflow is running.

wsmirnow committed:
Azure transcription service output format option

Azure transcription service should be able to produce vtt and srt files based on the service configuration.

wsmirnow committed:
Fixed and improved tests

... for Azure transcription service

wsmirnow committed:
Code cleanup

wsmirnow committed:
Improve handling of Azure transcription not found

wsmirnow committed:
Reimplementation of Microsoft Azure transcription service

The new implementation of Microsoft Azure transcription service uses the
Microsoft Azure Speech Services and Storage Services REST API to get the
work done. Therefore we not depend on Microsoft SDK, which depends on
GStreamer.

wsmirnow committed:
OC 15 rebase

wsmirnow committed:
Azure transcription service rebase to develop

wsmirnow committed:
Azure transcription service tests improved

wsmirnow committed:
Azure transcription service configuration documentation

wsmirnow committed:
Azure transcription service improvements

wsmirnow committed:
Azure transcription service documentation

wsmirnow committed:
Azure transcription service release notice

wsmirnow commented:
Below are some screenshots from generated transcriptions and workflow properties of the started attach-transcription job.

This image shows the attached transcription and appended tags generated by the workflow in the documentation
![image](https://github.com/opencast/opencast/assets/26736/83c3df0f-cb85-400c-95f4-42ffd5bc7ca0)

Here you can see the workflow properties
![image](https://github.com/opencast/opencast/assets/26736/6899a5da-1a24-43f2-bb6d-1d03f0e8e941)

Next screenshots shows you the same episode in editor with enabled subtitle editor
![image](https://github.com/opencast/opencast/assets/26736/804678f4-222e-40c8-b1d8-68d13af580f0)
![image](https://github.com/opencast/opencast/assets/26736/3e240193-8430-4ea7-84ec-720f9abd32f8)

wsmirnow committed:
Fix tag templating in Azure attach-transcription WOH

On empty value, the template string is added as tag. We want to make use of Freemarker template engine and define a default value.

wsmirnow committed:
Microsoft Azure Attach WOH should handle variables

The target-tags can make use of workflow properties. target-flavor should be able to do same.

wsmirnow commented:
The code is tested in production at ETH for some semesters now and it fixes some serious issues from the old implementation. I would like to see it merged. Is someone willing to review it?

KatrinIhler committed:
Update microsoftazure.md

KatrinIhler committed:
Update microsoft-azure-start-transcription-woh.md

KatrinIhler committed:
Update microsoft-azure-attach-transcription-woh.md

KatrinIhler committed:
Update org.opencastproject.transcription.microsoft.azure.MicrosoftAzureTranscriptionService.cfg

Set configuration to defaults, comment defaults out

KatrinIhler reviewed (approved):
Code looks structurally sound. I'll admit I didn't read every line but I skimmed every class and there was nothing unexpected in what I saw. Microsoft's API looks like a nightmare to implement but that's not on you. ;)  Looks like you made the best out of it.
The feature is also well-documented and has tests (yay!). I fixed some grammar and such in the documentation, hence the additional commits, but nothing that changes meaning.
Since this has been successfully running in production for quite a while, I feel comfortable merging this without trying it out. As soon as the merge conflict is fixed, this can go in, hopefully in time for OC 17. :)

wsmirnow committed:
Removed previous version of Microsoft Azure transcription service

wsmirnow committed:
Fixed documentation link

wsmirnow commented:
Thank you @KatrinIhler for the review. I've fixed the merge conflict, rebased the branch to develop and fixed related links in the admin guides. Please go ahead.
